File: OpenBurnBarCore/Sources/OpenBurnBarCore/Views/Substrate/Families/ConstellationFamily.swift
import Foundation

/// Constellation family substrate registry — the only catalog edit point for the
/// four Constellation bespoke authors. Ported from imaginethat
/// `glyph/stage/styles/constellation/`.
enum ConstellationFamily {
    static let styles: [SubstrateDescriptor] = [
        SubstrateDescriptor(
            id: "constellation.starfire", family: .constellation,
            label: "Stellar Plasma", hint: "twinkle",
            accent: RGBA(r: 131/255, g: 142/255, b: 1.0), accent2: RGBA(r: 150/255, g: 200/255, b: 1.0),
            make: { StarfireSubstrate() }),
        SubstrateDescriptor(
            id: "constellation.starsapphire", family: .constellation,
            label: "Cut Star Sapphire", hint: "facets",
            accent: RGBA(r: 120/255, g: 170/255, b: 1.0), accent2: RGBA(r: 180/255, g: 220/255, b: 1.0),
            make: { StarSapphireSubstrate() }),
        SubstrateDescriptor(
            id: "constellation.stellarium", family: .constellation,
            label: "Drawn Constellation", hint: "lines",
            accent: RGBA(r: 150/255, g: 170/255, b: 1.0), accent2: RGBA(r: 200/255, g: 215/255, b: 1.0),
            make: { StellariumSubstrate() }),
        SubstrateDescriptor(
            id: "constellation.rimefrost", family: .constellation,
            label: "Dendritic Frost", hint: "frost",
            accent: RGBA(r: 190/255, g: 215/255, b: 1.0), accent2: RGBA(r: 225/255, g: 240/255, b: 1.0),
            make: { RimefrostSubstrate() }),
    ]
}

File: windows/particles/OpenBurnBar.Particles/Substrates/Families/ConstellationFamily.cs
using OpenBurnBar.Particles.Model;

namespace OpenBurnBar.Particles.Substrates.Families;

/// <summary>
/// Constellation family registry — C# port of Swift <c>Families/ConstellationFamily.swift</c>.
/// The star-field / link family: Stellar Plasma (Starfire, landed in #1202) + Cut Star
/// Sapphire + Drawn Constellation + Dendritic Frost (the remaining three landed with the
/// Volumetric lane, #1213). Ids / labels / accents mirror the Swift descriptors
/// byte-for-byte.
/// </summary>
public static class ConstellationFamily
{
    public static readonly SubstrateDescriptor[] Styles =
    {
        new SubstrateDescriptor(
            id: "constellation.starfire", family: SubstrateFamily.Constellation,
            label: "Stellar Plasma", hint: "twinkle",
            accent: new Rgba(131 / 255.0, 142 / 255.0, 1.0),
            accent2: new Rgba(150 / 255.0, 200 / 255.0, 1.0),
            make: () => new StarfireSubstrate()),
        new SubstrateDescriptor(
            id: "constellation.starsapphire", family: SubstrateFamily.Constellation,
            label: "Cut Star Sapphire", hint: "facets",
            accent: new Rgba(120 / 255.0, 170 / 255.0, 1.0),
            accent2: new Rgba(180 / 255.0, 220 / 255.0, 1.0),
            make: () => new StarSapphireSubstrate()),
        new SubstrateDescriptor(
            id: "constellation.stellarium", family: SubstrateFamily.Constellation,
            label: "Drawn Constellation", hint: "lines",
            accent: new Rgba(150 / 255.0, 170 / 255.0, 1.0),
            accent2: new Rgba(200 / 255.0, 215 / 255.0, 1.0),
            make: () => new StellariumSubstrate()),
        new SubstrateDescriptor(
            id: "constellation.rimefrost", family: SubstrateFamily.Constellation,
            label: "Dendritic Frost", hint: "frost",
            accent: new Rgba(190 / 255.0, 215 / 255.0, 1.0),
            accent2: new Rgba(225 / 255.0, 240 / 255.0, 1.0),
            make: () => new RimefrostSubstrate()),
    };
}
